algorithms/kernel/pca/oneapi/pca_dense_correlation_online_kernel_ucapi_impl.i

```diff
@@ -0,0 +1,146 @@
+    }
+    if (s) s = copyCovarianceResultToPartialResult(parameter->covariance->getPartialResult().get(), partialResult);
+
+    return s;
+}
+
+template <typename algorithmFPType>
+services::Status PCACorrelationKernelOnlineUCAPI<algorithmFPType>::copyCovarianceResultToPartialResult(
+    covariance::PartialResult * covariancePres, PartialResult<correlationDense> * partialResult)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.copy.to.partial.result);
+    services::Status s = copyIfNeeded(covariancePres->get(covariance::sum).get(), partialResult->get(sumCorrelation).get());
+    s |= copyIfNeeded(covariancePres->get(covariance::nObservations).get(), partialResult->get(nObservationsCorrelation).get());
+    s |= copyIfNeeded(covariancePres->get(covariance::crossProduct).get(), partialResult->get(crossProductCorrelation).get());
+    return s;
+}
+
+template <typename algorithmFPType>
+services::Status PCACorrelationKernelOnlineUCAPI<algorithmFPType>::copyIfNeeded(data_management::NumericTable * src,
+                                                                                  data_management::NumericTable * dst)
+{
+    if (src == dst) return services::Status();
+
+    services::Status status;
+
+    BlockDescriptor<algorithmFPType> srcBlock;
+    BlockDescriptor<algorithmFPType> dstBlock;
+
+    {
+        status |= src->getBlockOfRows(0, src->getNumberOfRows(), readOnly, srcBlock);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        status |= dst->getBlockOfRows(0, dst->getNumberOfRows(), writeOnly, dstBlock);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    const size_t nRows = dst->getNumberOfRows();
+    const size_t nCols = dst->getNumberOfColumns();
+    const size_t nDataElements = nRows * nCols;
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+    context.copy(dstBlock.getBuffer(), 0, srcBlock.getBuffer(), 0, nDataElements, &status);
+
+    src->releaseBlockOfRows(srcBlock);
+    dst->releaseBlockOfRows(dstBlock);
+
+    return status;
+}
+
+template <typename algorithmFPType>
+Status PCACorrelationKernelOnlineUCAPI<algorithmFPType>::finalize(PartialResult<correlationDense> * partialResult,
+                                                                const OnlineParameter<algorithmFPType, correlationDense> * parameter,
+                                                                data_management::NumericTable & eigenvectors,
+                                                                data_management::NumericTable & eigenvalues)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(finalize);
+    services::Status s;
+    {
+        DAAL_ITTNOTIFY_SCOPED_TASK(compute.covariance.finalize);
+        s = parameter->covariance->finalizeCompute();
+        if (!s) return s;
+    }
+
+    data_management::NumericTablePtr correlation = parameter->covariance->getResult()->get(covariance::covariance);
+    _host_impl->computeCorrelationEigenvalues(*correlation, eigenvectors, eigenvalues);
```

Comment:
Was the decision made to leave eigensolver on CPU?

Reply:
GPU impl is so slow.